lib/srv/db/proxy_test.go

```diff
@@ -71,3 +74,83 @@ func TestProxyProtocolMySQL(t *testing.T) {
 	require.NoError(t, err)
 	require.NoError(t, mysql.Close())
 }
+
+// TestProxyClientDisconnectDueToIdleConnection ensures that idle clients will be disconnected.
+func TestProxyClientDisconnectDueToIdleConnection(t *testing.T) {
+	const (
+		idleClientTimeout             = time.Minute
+		connMonitorDisconnectTimeBuff = time.Second * 5
+	)
+
+	ctx := context.Background()
+	testCtx := setupTestContext(ctx, t, withSelfHostedMySQL("mysql"))
+	go testCtx.startHandlingConnections()
+
+	testCtx.createUserAndRole(ctx, t, "alice", "admin", []string{"root"}, []string{types.Wildcard})
+	seConfigClientIdleTimoutAndDisconnectExpiredCert(t, testCtx.authServer, idleClientTimeout)
+
+	proxy, err := multiplexer.NewTestProxy(testCtx.mysqlListener.Addr().String())
+	require.NoError(t, err)
+	t.Cleanup(func() { proxy.Close() })
+	go proxy.Serve()
+
+	mysql, err := testCtx.mysqlClientWithAddr(proxy.Address(), "alice", "mysql", "root")
+	require.NoError(t, err)
+
+	err = mysql.Ping()
+	require.NoError(t, err)
+
+	testCtx.clock.Advance(idleClientTimeout + connMonitorDisconnectTimeBuff)
+
+	requireEvent(t, testCtx, events.DatabaseSessionStartCode)
+	requireEvent(t, testCtx, events.ClientDisconnectCode)
+	err = mysql.Ping()
+	require.Error(t, err)
+}
+
+// TestProxyClientDisconnectDueToCertExpiration ensures that if the DisconnectExpiredCert cluster flag is enabled
+// clients will be disconnected after cert expiration.
+func TestProxyClientDisconnectDueToCertExpiration(t *testing.T) {
+	const (
+		ttlClientCert = time.Hour
+	)
+
+	ctx := context.Background()
+	testCtx := setupTestContext(ctx, t, withSelfHostedMySQL("mysql"))
+	go testCtx.startHandlingConnections()
+
+	testCtx.createUserAndRole(ctx, t, "alice", "admin", []string{"root"}, []string{types.Wildcard})
+	seConfigClientIdleTimoutAndDisconnectExpiredCert(t, testCtx.authServer, time.Hour*24)
+
+	proxy, err := multiplexer.NewTestProxy(testCtx.mysqlListener.Addr().String())
+	require.NoError(t, err)
+	t.Cleanup(func() { proxy.Close() })
+	go proxy.Serve()
+
+	mysql, err := testCtx.mysqlClientWithAddr(proxy.Address(), "alice", "mysql", "root")
+	require.NoError(t, err)
+
+	err = mysql.Ping()
+	require.NoError(t, err)
+
+	testCtx.clock.Advance(ttlClientCert)
+
+	requireEvent(t, testCtx, events.DatabaseSessionStartCode)
+	requireEvent(t, testCtx, events.ClientDisconnectCode)
+	err = mysql.Ping()
+	require.Error(t, err)
+}
+
+func seConfigClientIdleTimoutAndDisconnectExpiredCert(t *testing.T, auth *auth.Server, timeout time.Duration) {
```

Comment:
What does `se` prefix mean here? Is this a typo for `set` maybe?

Reply:
yes it is a typo, it supposed to bet `set...` fixed.